qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -0,0 +1,436 @@
+
+        # Setting frequency range
+        self.add_parameter('start',
+                           label='Start Frequency',
+                           get_cmd='SENS:FREQ:STAR?',
+                           get_parser=float,
+                           set_cmd='SENS:FREQ:STAR {}',
+                           unit='',
+                           vals=Numbers(min_value=min_freq,max_value=max_freq))
+        self.add_parameter('stop',
+                           label='Stop Frequency',
+                           get_cmd='SENS:FREQ:STOP?',
+                           get_parser=float,
+                           set_cmd='SENS:FREQ:STOP {}',
+                           unit='',
+                           vals=Numbers(min_value=min_freq,max_value=max_freq))
+
+        # Number of points in a sweep
+        self.add_parameter('points',
+                           label='Points',
+                           get_cmd='SENS:SWE:POIN?',
+                           get_parser=int,
+                           set_cmd='SENS:SWE:POIN {}',
+                           unit='',
+                           vals=Numbers(min_value=1, max_value=100001))
+
+        # Electrical delay
+        self.add_parameter('electrical_delay',
+                           label='Electrical Delay',
+                           get_cmd='CALC:CORR:EDEL:TIME?',
+                           get_parser=float,
+                           set_cmd='CALC:CORR:EDEL:TIME {:.6e}',
+                           unit='s',
+                           vals=Numbers(min_value=0, max_value=100000))
+
+        # Sweep Time
+        self.add_parameter('sweep_time',
+                           label='Time',
+                           get_cmd='SENS:SWE:TIME?',
+                           get_parser=float,
+                           unit='s',
+                           vals=Numbers(0,1e6))
+        # Sweep Mode
+        self.add_parameter('sweep_mode',
+                           label='Mode',
+                           get_cmd='SENS:SWE:MODE?',
+                           set_cmd='SENS:SWE:MODE {}',
+                           vals=Enum("HOLD", "CONT", "GRO", "SING"))
+
+        # Traces
+        # Note: These will be accessed through the traces property which updates
+        # the channellist to include only active trace numbers
+        self._traces = ChannelList(self, "PNATraces", PNATrace)
+        self.add_submodule("traces", self._traces)
+        # Add shortcuts to trace 1
+        trace1 = PNATrace(self, "tr1", 1)
+        for param in trace1.parameters.values():
+            self.parameters[param.name] = param
+        # By default we should also pull any following values from this trace
+        self.write("CALC:PAR:MNUM 1")
+
+        # Set auto_sweep parameter
+        # If we want to return multiple traces per setpoint without sweeping
+        # multiple times, we should set this to false
+        self._auto_sweep = True
```

Comment:
why do you wrap `_auto_sweep` into a parameter, or why do you have a private attribute AND a parameter? So far as I see, the following will work just fine without the need for `_auto_sweep` and `_set_auto_sweep`:
```python
self.add_parameter('auto_sweep',
    label='Auto Sweep',
    set_cmd=None,
    get_cmd=None,
    vals=Bool(),
    initial_value=True)
```

Reply:
Sure :)

I guess I haven't really seen any docs suggesting the best way to handle local parameters? If this is the preferred way, I'm happy to go with it 👍

Reply:
great!

well, `Parameter`s are always cached: you can call `get_latest` to get the latest value of the parameter without accessing the instrument.